keps/sig-auth/2907-secrets-store-csi-driver/README.md

```diff
@@ -0,0 +1,368 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The [secrets-store-csi-driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver) project provides a portable method for applications to consume secrets from external secret APIs through the filesystem. This effort was added to the `sig-auth` subproject in February 2020 and currently there are providers for Azure, AWS, GCP, and HashiCorp Vault. This KEP intends to cover making the core functionality of the driver GA.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- Signal the stability of the driver interface and implementation for the core task of making secrets available to pod filesystem.
+
+### Non-Goals
+
+- Extending the Kubernetes Secret object
+- Introduce a new Kubernetes type
+
+## Proposal
+
+This project introduces a new Container Storage Interface (CSI) driver for fetching secrets and writing to a `tmpfs` mount in the Pod filesystem. The driver is deployed as a `DaemonSet`. A new Custom Resource Definition (CRD) called a `SecretProviderClass` is introduced that informs the driver of which external secret storage API to contact and how to map the secrets from those APIs to file paths. The driver communicates with the external secret provider processes through a gRPC interface over a Unix Domain Socket.
+
+### User Stories (Optional)
+
+#### Story 1
+
+1. Application reads secret from filesystem on startup
+2. Application watches secret for rotation
+3. Application Pod YAML remains unchanged and works across secret providers
+
+### Notes/Constraints/Caveats (Optional)
+
+Since the proposal is a storage driver, native support for presenting secrets to a process through environment variables is not possible.
```

Comment:
Do we want to mention the optional `Sync as Kubernetes secret` feature can address this?

Reply:
Added!